frontend/src/stores/useBiolink.js

```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect } from 'react';
+import _ from 'lodash';
+import strings from '~/utils/strings';
+
+const baseClass = 'biolink:BiologicalEntity';
+
+export default function useBiolink() {
+  const [biolink, setBiolink] = useState(null);
+  const [concepts, setConcepts] = useState([]);
+  const [descendancies, setDescendancies] = useState({});
+  const [predicates, setPredicates] = useState([]);
+
+  function getEdgePredicates() {
+    const newPredicates = Object.entries(biolink.slots).map(([identifier, predicate]) => ({
+      predicate: strings.edgeFromBiolink(identifier),
+      domain: strings.nodeFromBiolink(predicate.domain),
+      range: strings.nodeFromBiolink(predicate.range),
+    }));
+    return newPredicates;
+  }
+
+  /**
+   * Given a biolink class, build a list of parent classes
+   * Ex: Given 'gene' as an input this function will return:
+   *
+   * [ "gene", "gene_or_gene_product", "macromolecular_machine",
+   *   "genomic_entity", "molecular_entity", "biological_entity",
+   *   "named_thing" ]
+   *
+   * @param {string} className - a biolink class
+   * @param {object} biolinkClasses - object of all biolink classes
+   * @returns {string[]} list of related biolink classes
+   */
+  function getDescendants(childClass, classes) {
+    let currentClass = childClass;
+    const descendantList = [currentClass];
+    // Repeat until we hit the top of the classes
+    while (
+      classes[currentClass] &&
+      classes[currentClass].is_a
+    ) {
+      // reassign current type to parent type
+      currentClass = classes[currentClass].is_a;
+      descendantList.push(currentClass);
+    }
+    return descendantList;
+  }
+
+  /**
+   * Get a list of descendants for all biolink classes
+   * @param {object} biolinkClasses - object of all biolink classes
+   * @returns {{}} Object with classes as keys and descendant lists as values
+   */
+  function getAllDescendancies(biolinkClasses) {
+    const newHierarchies = Object.keys(biolinkClasses).reduce((obj, item) => {
+      let hierarchy = getDescendants(item, biolinkClasses);
```

Comment:
```suggestion
      let descendants = getDescendants(item, biolinkClasses);
```

Reply:
Updated in 660b8dbce25b9f08b63354a31158a40743511447

Reply:
By "ascendants" I think you mean "ancestors"? English is bonkers, so "ascendant" means something else entirely.

Reply:
According to dictionary.com, ascendant can mean ancestor. https://www.dictionary.com/browse/ascendant

Reply:
Changed in ancestors in b0ddeeb80bf7ac10835a57ec53a532bba97e1b7d